fly/commands/clear_resource_cache.go

```diff
@@ -0,0 +1,57 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/fly/commands/internal/flaghelpers"
+	"github.com/concourse/concourse/fly/rc"
+	"github.com/concourse/concourse/fly/ui"
+	"github.com/fatih/color"
+)
+
+type ClearResourceCacheCommand struct {
+	Resource flaghelpers.ResourceFlag `short:"r" long:"resource-type" required:"true" value-name:"PIPELINE/RESOURCE-TYPE" description:"Name of a resource to invalidate"`
+	Version  *atc.Version             `short:"f" long:"from" required:"false" value-name:"VERSION" description:"Version of the resource cache to invalidate, e.g. digest:sha256@..."`
```

Comment:
```suggestion
	Version  *atc.Version             `long:"version" required:"false" value-name:"VERSION" description:"Version of the resource cache to clear, e.g. ref:abcdef..."`
```

I removed the short form as there's ambiguity with `-v` for `--version`. I think in the long run we should redesign the CLI to take these values as arguments so we don't have so many overlapping short flags. :thinking: But that's a topic for another day. For now let's just have the long flag name.